### PR TITLE
REGR: Fix IntegerArray unary ops regression

### LINKED ISSUE #36063: BUG: Unary minus raises for series with Int64Dtype
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas (commit 73c1d32)

---

#### Code Sample, a copy-pastable example

```python
>>> import pandas as pd
>>> pd.__version__
'1.1.1'
>>> s = pd.Series([1, 2, 3], dtype="Int64")
>>> -s
Traceback (most recent call last):
  File "...", line 1, in <module>
    -s
  File ".../lib/python3.8/site-packages/pandas/core/generic.py", line 1297, in __neg__
    arr = operator.neg(values)
TypeError: bad operand type for unary -: 'IntegerArray'
```

#### Problem description

I cannot negate series with Int64Dtype. This was possible in previous versions (but it returned object dtype).

#### Expected Output

```
>>> import pandas as pd
>>> pd.__version__
'1.0.5'
>>> s = pd.Series([1, 2, 3], dtype="Int64")
>>> -s
0    -1
1    -2
2    -3
dtype: object
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : f2ca0a2665b2d169c97de87b8e778dbed86aea07
python           : 3.8.3.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 19.6.0
Version          : Darwin Kernel Version 19.6.0: Thu Jun 18 20:49:00 PDT 2020; root:xnu-6153.141.1~1/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.1.1
numpy            : 1.19.1
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.1.1
setuptools       : 47.3.1
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : 7.15.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by dsaxton**
- [x] closes #36063
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/arrays/integer.py [hunk 1]
```diff
@@ -364,6 +364,9 @@ def __init__(self, values: np.ndarray, mask: np.ndarray, copy: bool = False):
             )
         super().__init__(values, mask, copy=copy)
 
+    def __neg__(self):
+        return type(self)(0 - self._data, self._mask)
```
Comment (jbrockmendel) [inline pandas/core/arrays/integer.py:368]: can you add a test for this directly in the IntegerArray tests

Comment (jbrockmendel) [reply pandas/core/arrays/integer.py:368]: and probably put this next to sum or _create_arithmetic_method?

could do `__pos__` and `__abs__` while at it?

File: pandas/tests/series/test_arithmetic.py [hunk 2]
```diff
@@ -687,3 +687,27 @@ def test_datetime_understood(self):
         result = series - offset
         expected = pd.Series(pd.to_datetime(["2011-12-26", "2011-12-27", "2011-12-28"]))
         tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize("dtype", ["Int64", "Int32", "Int16", "Int8"])
```
Comment (jbrockmendel) [inline pandas/tests/series/test_arithmetic.py:691]: series unary ops are tested in tests.series.test_operators.TestSeriesUnaryOps

File: pandas/core/generic.py [hunk 3]
```diff
@@ -1408,16 +1408,18 @@ def __neg__(self):
 
     def __pos__(self):
         values = self._values
-        if is_bool_dtype(values):
-            arr = values
-        elif (
-            is_numeric_dtype(values)
+        if (
+            is_bool_dtype(values)
+            or is_numeric_dtype(values)
```
Comment (dsaxton) [inline pandas/core/generic.py:1413]: The is_numeric_dtype part of this is needed for the fix, the rest is cleanup I figure might as well do while we're here

File: pandas/core/arrays/integer.py [hunk 4]
```diff
@@ -364,6 +364,15 @@ def __init__(self, values: np.ndarray, mask: np.ndarray, copy: bool = False):
...
         super().__init__(values, mask, copy=copy)
 
+    def __neg__(self):
+        return type(self)(-self._data, self._mask)
+
+    def __pos__(self):
+        return self
+
+    def __abs__(self):
+        return type(self)(np.sign(self._data) * self._data, self._mask)
```
Comment (jreback) [inline pandas/core/arrays/integer.py:374]: np.abs(self._data) no?

File: pandas/core/generic.py [hunk 5]
```diff
@@ -1408,16 +1408,18 @@ def __neg__(self):
...
             or is_timedelta64_dtype(values)
             or is_object_dtype(values)
         ):
-            arr = operator.pos(values)
+            arr = values
```
Comment (jreback) [inline pandas/core/generic.py:1417]: why did you remove this? we need to dispatch to the object itself in case it needs to actually take action

File: pandas/tests/arrays/integer/test_arithmetic.py [hunk 6]
```diff
@@ -261,3 +261,41 @@ def test_reduce_to_float(op):
...
+    ],
+)
+def test_unary_minus_nullable_int(dtype, source, target):
+    arr = pd.array(source, dtype=dtype)
+    result = -arr
+    expected = pd.array(target, dtype=dtype)
+    tm.assert_extension_array_equal(result, expected)
+
+
+@pytest.mark.parametrize("dtype", ["Int64", "Int32", "Int16", "Int8"])
```
Comment (jreback) [inline pandas/tests/arrays/integer/test_arithmetic.py:282]: can you use the any_nullable_int_dtype fixture instead

File: pandas/core/generic.py [hunk 7]
```diff
@@ -1408,16 +1408,18 @@ def __neg__(self):
...
         else:
-            raise TypeError(f"Unary plus expects numeric dtype, not {values.dtype}")
+            raise TypeError(
+                "Unary plus expects bool, numeric, timedelta, "
+                f"or object dtype, not {values.dtype}"
```
Comment (jreback) [inline pandas/core/generic.py:1421]: are we actually testing this message?

File: pandas/tests/series/test_operators.py [hunk 8]
```diff
@@ -536,3 +536,42 @@ def test_invert(self):
...
+            ([-1, 0, 1], [1, 0, -1]),
+        ],
+    )
+    def test_unary_minus_nullable_int(self, dtype, source, target):
+        s = pd.Series(source, dtype=dtype)
+        result = -s
+        expected = pd.Series(target, dtype=dtype)
+        tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize("dtype", ["Int64", "Int32", "Int16", "Int8"])
```
Comment (jreback) [inline pandas/tests/series/test_operators.py:555]: use the fixture

File: pandas/tests/series/test_operators.py [hunk 9]
```diff
@@ -536,3 +536,42 @@ def test_invert(self):
...
+    @pytest.mark.parametrize(
+        "source", [[1, 2, 3], [1, 2, None], [-1, 0, 1]],
+    )
+    def test_unary_plus_nullable_int(self, dtype, source):
+        expected = pd.Series(source, dtype=dtype)
+        result = +expected
+        tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize("dtype", ["Int64", "Int32", "Int16", "Int8"])
```
Comment (jreback) [inline pandas/tests/series/test_operators.py:564]: use the fixture

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/generic.py (hunk 5 above)
Comment (dsaxton) [reply pandas/core/generic.py:1417]: This was left over from before I added `__pos__`, will fix now

File: pandas/core/generic.py (hunk 7 above)
Comment (dsaxton) [reply pandas/core/generic.py:1421]: Yes, the updated test is in /frame/test_operators.py

File: pandas/tests/arrays/integer/test_arithmetic.py (hunk 6 above)
Comment (dsaxton) [reply pandas/tests/arrays/integer/test_arithmetic.py:282]: We can't because these are signed integers, could create another fixture for signed if that makes sense

File: pandas/conftest.py [hunk 10]
```diff
@@ -1055,6 +1055,19 @@ def any_nullable_int_dtype(request):
     return request.param
 
 
+@pytest.fixture(params=tm.SIGNED_EA_INT_DTYPES)
```
Comment (jreback) [inline pandas/conftest.py:1058]: great!

**Review by jreback — APPROVED** [review]
very nice!

**Comment by dsaxton** [comment]
@jreback @simonjayhawkins Should this change be backported?

**Comment by simonjayhawkins** [comment]
will do it after merged the two open ones. if meeks... fails can't retrigger and need to do it manually

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x
